src/widgets/editor.py

```diff
@@ -78,35 +96,62 @@ def updatePreferences(self,*args):
     
     def new(self):
         
-        self._filename = ''
+        self.update_filename('')
         self.set_text('')
 
     def open(self):
         
         fname,_ = QFileDialog.getOpenFileName(self,filter=self.EXTENSIONS)
         if fname is not '':
             self.load_from_file(fname)
-            
+
     def load_from_file(self,fname):
         
         self.set_text_from_file(fname)
-        self._filename = fname
-    
+        self.update_filename(fname)
+
     def save(self):
         
         if self._filename is not '':
             with open(self._filename,'w') as f:
                 f.write(self.get_text_with_eol())
         else:
             self.save_as()
-        
+
     def save_as(self):
         
         fname,_ = QFileDialog.getSaveFileName(self,filter=self.EXTENSIONS)
         if fname is not '':
-             with open(fname,'w') as f:
+            with open(fname,'w') as f:
                 f.write(self.get_text_with_eol())
-                self._filename = fname
+                self.update_filename(fname)
+
+    def _update_filewatcher(self):
+        if self._watched_file and ((self._watched_file != self._filename) or not self._autoreload_enabled):
+            logging.debug("unwatching file: {}".format(self._watched_file))
+            self._file_watcher.removePath(self._watched_file)
+            self._watched_file = None
+        if self._autoreload_enabled and self._filename and self._filename != self._watched_file:
+            logging.debug("watching file: {}".format(self._filename))
+            self._watched_file = self._filename
+            self._file_watcher.addPath(self._filename)
+
+    def update_filename(self, fname):
+        self._filename = fname
+        self._update_filewatcher()
+
+    # callback triggered by QFileSystemWatcher
```

Comment:
I got kind of lost - why not directly connect flieChanged to a lambda emitting triggerRender? The logic should be already handled elsewhere (i.e. in updateSettings)

Reply:
I removed the extra logic, which shouldn't be necessary. The code is still two lines though since it updates the editor text and triggers a re-render.